File: eAnimalShelter/UI/eanimalshelter_desktop/lib/models/adoption_request.dart
import 'animal.dart';
import 'user.dart';

class AdoptionRequest {
  int adoptionRequestId;

  int userId;
  int animalId;

  String? userName;
  String? animalName;

  DateTime requestDate;

  String housingType;
  String experienceWithPets;
  int householdMembers;

  String? additionalNotes;

  int status;

  String? adminComment;

  User? user;
  Animal? animal;

  AdoptionRequest({
    required this.adoptionRequestId,
    required this.userId,
    required this.animalId,
    this.userName,
    this.animalName,
    required this.requestDate,
    required this.housingType,
    required this.experienceWithPets,
    required this.householdMembers,
    this.additionalNotes,
    required this.status,
    this.adminComment,
    this.user,
    this.animal,
  });

  factory AdoptionRequest.fromJson(
      Map<String, dynamic> json) {
    return AdoptionRequest(
      adoptionRequestId:
          json["adoptionRequestId"],
      userId: json["userId"],
      animalId: json["animalId"],
      userName: json["userName"],
      animalName: json["animalName"],
      requestDate:
          DateTime.parse(json["requestDate"]),
      housingType:
          json["housingType"] ?? "",
      experienceWithPets:
          json["experienceWithPets"] ?? "",
      householdMembers:
          json["householdMembers"] ?? 0,
      additionalNotes:
          json["additionalNotes"],
      status: json["status"] ?? 0,
      adminComment:
          json["adminComment"],
      user: json["user"] != null
          ? User.fromJson(json["user"])
          : null,
      animal: json["animal"] != null
          ? Animal.fromJson(json["animal"])
          : null,
    );
  }
}

File: eAnimalShelter/eAnimalShelter.Services/Database/AdoptionRequest.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using eAnimalShelter.Model.Enums;

namespace eAnimalShelter.Services.Database
{
    public class AdoptionRequest
    {
        [Key]
        public int AdoptionRequestId { get; set; }
        public int UserId { get; set; }
        [ForeignKey("UserId")]
        public User User { get; set; } = null!;
        public int AnimalId { get; set; }
        [ForeignKey("AnimalId")]
        public Animal Animal { get; set; } = null!;

        public DateTime RequestDate { get; set; }
        
        public string HousingType { get; set; } = string.Empty;

        public string ExperienceWithPets { get; set; } = string.Empty;

        public int HouseholdMembers { get; set; }

        public string? AdditionalNotes { get; set; }

        public AdoptionRequestStatus Status { get; set; }
            = AdoptionRequestStatus.Pending;
        public int? ReviewedBy { get; set; }
        [ForeignKey("ReviewedBy")]
        public User? Reviewer { get; set; }
        public DateTime? ReviewDate { get; set; }
        public string? AdminComment { get; set; }
    }

}